sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -553,10 +575,29 @@ object DottyPlugin extends AutoPlugin {
       dottyVersion,
       fullLoader,
       libraryLoader,
-      Array(dottyLibrary, scalaLibrary),
+      libraryJars,
       compiler,
       all.toArray,
       None)
+  }
+}
+
+private class FilteringClassLoader(parent: ClassLoader) extends ClassLoader(parent) {
+  private val exludePackages = List(
```

Comment:
Listing the packages to include rather than the packages to exclude like you did in the first commit seems safer to me. (for example, you're not excluding "dotty." which could be loaded by a future version of sbt). However in that commit you had to include all the JDK prefixes ("java.", "sun.", ...). Perhaps this can be avoided by replicating what CompilerClassLoader did before this PR: use `null` as a parent for FilteringClassLoader so `super.loadClass` will delegate to the system JDK classloader, and call loadClass on the sbt loader explicitly when needed instead.